### PR TITLE
Add basic functions to Lesson and Task related classes

### PR DESCRIPTION
**Pull request by Roycius**
## Changelog

* Added basic attributes and methods to the `Lesson` class
* Added basic attributes and methods to the `Task` class
* Added basic attributes and methods to the `LessonList` class
* Added basic attributes and methods to the `TaskList` class

### CODE REVIEW

File: src/main/java/seedu/duke/lesson/LessonList.java [hunk 1]
```diff
@@ -1,4 +1,36 @@
...
+
+    /**
+     * Removes the lesson of the given index from the lesson list.
+     *
+     * @param index the index of the lesson to be removed
+     */
+    public void deleteLessonFromList(int index) {
+        lessonList.remove(index);
+    }
+}
```
Comment (richwill28) [inline src/main/java/seedu/duke/lesson/LessonList.java:36]: No `\n` at the end of the file. Git coding convention violation.

File: src/main/java/seedu/duke/lesson/LessonList.java [hunk 2]
```diff
@@ -1,4 +1,36 @@
 package seedu.duke.lesson;
 
+import java.util.ArrayList;
+
 public class LessonList {
-}
+    private ArrayList<Lesson> lessonList = new ArrayList<>();
```
Comment (richwill28) [inline src/main/java/seedu/duke/lesson/LessonList.java:6]: Consider using `List<Lesson> lessonList = new ArrayList<>()` instead to improve flexibility.

Note: [Relevant info](https://stackoverflow.com/questions/2279030/type-list-vs-type-arraylist-in-java)

Comment (richwill28) [reply src/main/java/seedu/duke/lesson/LessonList.java:6]: Hmm.. actually might be better to just declare it first, without initializing. Initialization should be done inside the constructor.

File: src/main/java/seedu/duke/lesson/LessonList.java [hunk 3]
```diff
@@ -1,4 +1,36 @@
 package seedu.duke.lesson;
 
+import java.util.ArrayList;
+
 public class LessonList {
-}
+    private ArrayList<Lesson> lessonList = new ArrayList<>();
+
+    public LessonList() {
```
Comment (richwill28) [inline src/main/java/seedu/duke/lesson/LessonList.java:8]: Initialize here instead. For example, `lessonList = new ArrayList<>();`.

File: src/main/java/seedu/duke/lesson/LessonList.java [hunk 4]
```diff
@@ -1,4 +1,36 @@
...
+    /**
+     * Replaces the current lesson list with a given one.
+     *
+     * @param newLessonList the new lesson list given
+     */
+    public LessonList(ArrayList<Lesson> newLessonList) {
+        lessonList = newLessonList;
+    }
+
+    public void addLessonToList(Lesson newLesson) {
```
Comment (richwill28) [inline src/main/java/seedu/duke/lesson/LessonList.java:20]: `addLesson` may be a better name?

File: src/main/java/seedu/duke/lesson/LessonList.java [hunk 5]
```diff
@@ -1,4 +1,36 @@
...
+    public void clearLessonList() {
+        lessonList.clear();
+    }
+
+    /**
+     * Removes the lesson of the given index from the lesson list.
+     *
+     * @param index the index of the lesson to be removed
+     */
+    public void deleteLessonFromList(int index) {
```
Comment (richwill28) [inline src/main/java/seedu/duke/lesson/LessonList.java:33]: `deleteLesson` may be a better name?

File: src/main/java/seedu/duke/lesson/Lesson.java [hunk 6]
```diff
@@ -1,4 +1,35 @@
...
+        this.endTime = endTime;
+    }
+
+    public String getDayOfTheWeek() {
+        return dayOfTheWeek;
+    }
+
+    @Override
+    public String toString() {
+        return ("[L]" + "    "
```
Comment (richwill28) [inline src/main/java/seedu/duke/lesson/Lesson.java:30]: FYI, the brackets here are unnecessary.

File: src/main/java/seedu/duke/task/Task.java [hunk 7]
```diff
@@ -1,4 +1,60 @@
...
+     * Returns the symbol representing the done status of the task.
+     *
+     * @return the symbol representing whether the task is done
+     */
+    public String getDoneSymbol() {
+        return isDone ? "[X]" : "[ ]";
+    }
+
+    @Override
+    public String toString() {
```
Comment (richwill28) [inline src/main/java/seedu/duke/task/Task.java:50]: I think might be better to simplify it to this?

```java
return "[T]" + getDoneSymbol() + " " + "Title: " + title + " " + (information.isBlank() ? "" : "(Info: " + information + ")");
```

File: src/main/java/seedu/duke/task/TaskList.java [hunk 8]
```diff
@@ -1,4 +1,36 @@
 package seedu.duke.task;
 
+import java.util.ArrayList;
+
 public class TaskList {
+    private ArrayList<Task> taskList = new ArrayList<>();
```
Comment (richwill28) [inline src/main/java/seedu/duke/task/TaskList.java:6]: Similar issue with `lessonList`.

File: src/main/java/seedu/duke/task/TaskList.java [hunk 9]
```diff
@@ -1,4 +1,36 @@
 package seedu.duke.task;
 
+import java.util.ArrayList;
+
 public class TaskList {
+    private ArrayList<Task> taskList = new ArrayList<>();
+
+    public TaskList() {
```
Comment (richwill28) [inline src/main/java/seedu/duke/task/TaskList.java:8]: Similar issue with `lessonList`.

File: src/main/java/seedu/duke/task/TaskList.java [hunk 10]
```diff
@@ -1,4 +1,36 @@
...
+    /**
+     * Replaces the current task list with a given one.
+     *
+     * @param newTaskList the new task list given
+     */
+    public void loadTaskList(ArrayList<Task> newTaskList) {
+        taskList = newTaskList;
+    }
+
+    public void addTaskToList(Task newTask) {
```
Comment (richwill28) [inline src/main/java/seedu/duke/task/TaskList.java:20]: `addTask` may be a better name?

File: src/main/java/seedu/duke/task/TaskList.java [hunk 11]
```diff
@@ -1,4 +1,36 @@
...
+    public void clearTaskList() {
+        taskList.clear();
+    }
+
+    /**
+     * Removes the Task of the given index from the task list.
+     *
+     * @param index the index of the task to be removed
+     */
+    public void deleteTaskFromList(int index) {
```
Comment (richwill28) [inline src/main/java/seedu/duke/task/TaskList.java:33]: `deleteTask` may be a better name?

**Review by richwill28 — COMMENTED** [review]
Overall, looks okay to me. You might want to update the test cases and see if it works.

File: src/main/java/seedu/duke/lesson/LessonList.java (hunk 1 above)
Comment (Roycius) [reply src/main/java/seedu/duke/lesson/LessonList.java]: Thank you for all the suggestions! I have made the necessary adjustments to all the issues.

File: src/main/java/seedu/duke/task/Task.java [hunk 12]
```diff
@@ -52,9 +52,7 @@ public String toString() {
             return ("[T]" + getDoneSymbol() + " "
                     + "Title: " + title + " ");
         } else {
-            return ("[T]" + getDoneSymbol() + " "
-                    + "Title: " + title + " "
-                    + "(Info: " + information + ")");
+            return "[T]" + getDoneSymbol() + " " + "Title: " + title + " " + "(Info: " + information + ")";
         }
```
[the remark this answers is not in the document]
Comment (Roycius) [reply src/main/java/seedu/duke/task/Task.java]: Thank you for the suggestion once again, I have made the changes as per your suggestion.

**Pull request merged**